Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBufferProxy.cpp

```diff
@@ -105,6 +106,10 @@ void RemoteBufferProxy::unmap()
 
 void RemoteBufferProxy::destroy()
 {
+    if (m_destroyed)
+        return;
+
+    m_destroyed = true;
```

Comment:
I don't think this is right. There are 2 different notions:

1. An object exists, and is a real object, but it has had the WebGPU API destroy() function called on it. It fully exists from a JS point of view, but the object inside WebGPU.framework doesn't own a Metal object
2. The Javascript object gets garbage collected. There is no object from a JS point of view.

I don't think we should conflate these two things.

Reply:
Does it conflate the two? You can still call methods on the JS object, it shouldn't do anything. Do we need to maintain the connection to the GPU process for that object?

Reply:
I think, because the concept of destroy exists at the WGPU layer, we should hook up RemoteFooProxy::destroy() to wgpuFooDestroy(), to keep the same APIs as parallel as possible. If this turns out to be a perf/memory problem, then I think we should reevaluate then.